index.js

```diff
@@ -93,35 +97,135 @@ function addCurrency(event) {
     });
+
+    alert(`New currency ${newCurrencyCode} - ${newCurrencyName} added with rate ${newRate}`);
   } else {
     alert(`Currency code ${newCurrencyCode} already exists.`);
   }
-  console.log(currencies); // for check if its added
+
+  console.log(currencies); // for check if it's added
 }
 
 // Update currency function
 function updateCurrency(event) {
   event.preventDefault(); // Prevent form submission
 
   const updatedRate = parseFloat(document.getElementById("update-rate").value);
-  const updatedCurrencyCode = document
-    .getElementById("to-currency-update")
-    .value.toUpperCase();
+  const updatedCurrencyCode = document.getElementById("to-currency-update").value.toUpperCase();
 
-  // Check if the currency code exists in the currencies object
-  if (currencies.rates[updatedCurrencyCode]) {
-    // Update the rate of the currency
-    currencies.rates[updatedCurrencyCode].rate = updatedRate;
+  // Find the currency to update
+  const currencyToUpdate = currencies.rates.find((rate) => rate.code === updatedCurrencyCode);
 
+  if (currencyToUpdate) {
+    currencyToUpdate.rate = updatedRate;
     alert(`Rate for currency ${updatedCurrencyCode} updated to ${updatedRate}`);
+  } else {
+    alert(`Currency code ${updatedCurrencyCode} does not exist.`);
   }
 
-  console.log(currencies); // for check if its updated
+  console.log(currencies); // for check if it's updated
 }
+
 function swapCurrencies() {
   const fromCurrency = selectFromElement.value;
   const toCurrency = selectToElement.value;
 
   selectFromElement.value = toCurrency;
   selectToElement.value = fromCurrency;
 }
+
+// Grid creation for list
+const gridList = document.getElementById("gridList");
+
+currencies.rates.forEach((rate) => {
+  const newGrid = document.createElement("div");
+  newGrid.classList.add("grid-item");
+  newGrid.textContent = `${rate.code} - ${rate.text} - Rate = ${rate.rate}`;
+  gridList.appendChild(newGrid);
+});
+// Search currency
+function findCurrencyRate(input) {
+  const query = input.trim().toUpperCase();
+
+  const foundRate = currencies.rates.find((rate) => {
+    return rate.code.toUpperCase().includes(query) || rate.text.toUpperCase().includes(query);
+  });
+
+  const searchResult = document.getElementById("searchResult");
+
+  if (foundRate) {
+    searchResult.innerHTML = `1 Euro = ${foundRate.rate} ${foundRate.text} (${
+      foundRate.code
+    })<br> 1 ${foundRate.code} = ${(1 / foundRate.rate).toFixed(4)} Euro `;
+  } else {
+    searchResult.innerHTML = `Currency rate for ${input} not found.`;
+  }
+}
+let isOpen;
+const openAndCloseAlert = (hour, minute, second, message) => {
+  const moment = new Date();
+  const targetTime = new Date();
+  targetTime.setHours(hour, minute, second, 0);
+  let timeDifference = targetTime - moment;
+
+  if (timeDifference < 0) {
+    targetTime.setDate(targetTime.getDate() + 1);
+    timeDifference = targetTime - moment;
+  }
+  if (moment.getHours() > 9 && moment.getHours() < 17) {
```

Comment:
You could simplify this as `isOpen = moment.getHours() > 9 && moment.getHours() < 17` (Can you guess why?)

But your way is also correct.

Reply:
Yeah its more make sense and short thanks for advice :)